client/allocwatcher/alloc_watcher.go

```diff
@@ -618,12 +624,12 @@ func (p *remotePrevAlloc) streamAllocDir(ctx context.Context, resp io.ReadCloser
 				return fmt.Errorf("error creating symlink: %w", err)
 			}
 
-			escapes, err := escapingfs.PathEscapesAllocDir(dest, "", hdr.Name)
-			if err != nil {
-				return fmt.Errorf("error evaluating symlink: %w", err)
-			}
-			if escapes {
-				return fmt.Errorf("archive contains symlink that escapes alloc dir")
+			for _, path := range []string{hdr.Name, hdr.Linkname} {
```

Comment:
If we're going to expand this check to cover checking the `hdr.Linkname`, we should cover that behavior with an additional test case on `TestPrevAlloc_StreamAllocDir_BadSymlink`. That should also make it clear why we've added this check, and why the check of `hdr.Name` at the top of the loop doesn't cover us.

Reply:
Good point, I added `TestPrevAlloc_StreamAllocDir_BadSymlink_Linkname` to cover the Linkname based escape test https://github.com/hashicorp/nomad/commit/8fe431da2f1b1646509fa1d94b85ab4416473945